components/expression/expression-renderer.js

```diff
@@ -244,180 +413,341 @@ genotet.ExpressionRenderer.prototype.drawGeneProfiles_ = function() {
     this.svgProfile_.selectAll('*').remove();
     return;
   }
+  else if (!this.data.options.showGeneLabels) {
+    this.PROFILE_MARGINS.LEFT = this.DEFAULT_PROFILE_MARGIN;
+  }
+  else {
+    this.PROFILE_MARGINS.LEFT = this.geneLabelWidth_ + this.LABEL_MARGIN_;
+  }
+
+  var heatmapData = this.data.matrix;
+  this.svgProfile_.selectAll('g').remove();
+  this.svgProfile_.attr('width', this.canvasWidth_);
+
+  var xScale = d3.scale.linear().range([
+    this.PROFILE_MARGINS.LEFT,
+    this.canvasWidth_ - this.PROFILE_MARGINS.RIGHT
+  ]).domain([0, heatmapData.conditionNames.length - 1]);
+  var yScale = d3.scale.linear().range([
+    this.profileHeight_ - this.PROFILE_MARGINS.BOTTOM,
+    this.PROFILE_MARGINS.TOP
+  ]).domain([heatmapData.valueMin, heatmapData.valueMax]);
+  var xAxis = d3.svg.axis()
+    .scale(xScale).orient('bottom');
+  var yAxis = d3.svg.axis()
+    .scale(yScale).orient('left');
+  this.svgProfile_
+    .append('svg:g').call(xAxis)
+    .classed('axis', true)
+    .classed('x', true)
+    .attr(
+      'transform',
+      'translate(0, ' + (this.profileHeight_ - this.PROFILE_MARGINS.BOTTOM) + ')'
+    );
+  this.svgProfile_
+    .append('svg:g').call(yAxis)
+    .classed('axis', true)
+    .attr('transform', 'translate(' + this.PROFILE_MARGINS.LEFT + ', 0)');
+
+  var colors = d3.scale.category20()
+    .range()
+    .concat(d3.scale.category20b().range())
+    .concat(d3.scale.category20c().range());
+  var profileContent = this.svgProfile_.append('g')
+    .attr(
+      'width',
+      this.canvasWidth_ - this.PROFILE_MARGINS.LEFT - this.PROFILE_MARGINS.RIGHT
+    )
+    .attr(
+      'height',
+      this.profileHeight_ - this.PROFILE_MARGINS.TOP - this.PROFILE_MARGINS.BOTTOM
+    );
+
+  var line = d3.svg.line()
+    .x(function(data, i) {
+      return xScale(i);
+    })
+    .y(function(data) {
+      return yScale(data);
+    })
+    .interpolate('linear');
+  this.data.profiles.forEach(function(path, i) {
+    var geneIndex = this.data.profiles[i].row;
+    var pathColor = colors[genotet.utils.hashString(heatmapData.geneNames[geneIndex]) % 60];
+    this.data.profiles[i] = {
+      geneName: heatmapData.geneNames[geneIndex],
+      row: geneIndex,
+      color: pathColor
+    };
+    profileContent.append('path')
+      .attr('d', line(heatmapData.values[geneIndex]))
+      .attr('stroke', pathColor)
+      .attr('fill', 'none')
+      .on('mouseover', function(value) {
+        this.data.profiles[i].container_= d3.event.target;
+        this.signal('pathHover', this.data.profiles[i]);
+      }.bind(this))
+      .on('mouseout', function(value) {
+        this.data.profiles[i].container_= d3.event.target;
+        this.signal('pathUnhover', this.data.profiles[i]);
```

Comment:
Please replace "path" by "profile". "path" is an implementation specific word. However data specifc words are preferred for readability.